### PR TITLE
[BIA-996] Need for all students with any enrollment

### PR DESCRIPTION
**Pull request by jleiva-gap**
Add scripts to create vie AllStudentDim (MSSQL 2,3.1,3.2,3.3 and Postgres 3.2, 3.3)
Add test cases.

### CODE REVIEW

File: src/EdFi.AnalyticsMiddleTier.DataStandard31/Base/MSSQL/0041-View-AllStudentSchoolDim-Create.sql [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            Student.StudentUniqueId
+            ,'-'
+            ,StudentSchoolAssociation.SchoolId
+            ) AS StudentSchoolKey
+        ,Student.StudentUniqueId AS StudentKey
+        ,CAST(StudentSchoolAssociation.SchoolId AS VARCHAR) AS SchoolKey
+        ,COALESCE(CAST(StudentSchoolAssociation.SchoolYear AS VARCHAR), 'Unknown') AS SchoolYear
+        ,Student.FirstName AS StudentFirstName
+        ,COALESCE(Student.MiddleName, '') AS StudentMiddleName
+        ,COALESCE(Student.LastSurname, '') AS StudentLastName
```
Comment (DavidJGapCR) [inline src/EdFi.AnalyticsMiddleTier.DataStandard31/Base/MSSQL/0041-View-AllStudentSchoolDim-Create.sql:35]: In DS 3.x student lastname can't be null, so we can remove the `coalesce` call here. And in DS 3.2 and 3.3.

Comment (jleiva-gap) [reply src/EdFi.AnalyticsMiddleTier.DataStandard31/Base/MSSQL/0041-View-AllStudentSchoolDim-Create.sql]: The column was updated to remove the coalesce.

**Comment by DavidJGapCR** [comment]
Data Standard 2:
. When I remove line 43 `OR StudentSchoolAssociation.ExitWithdrawDate > GETDATE()`
. When I comment out the COALESCE on line 48 `COALESCE(cast(StudentSchoolAssociation.ExitWithdrawDate AS VARCHAR(100)), '') AS ExitWithdrawDate`

Data Standard 3.x. (SQL Server and Postgres):
. When I comment out the COALESCE on lines: 44,54,55,56,57,58.
. When I remove the line 60: `OR StudentSchoolAssociation.ExitWithdrawDate > GETDATE()`

I think that if these scenarios are not covered on `StudentSchoolDim` either, we can create a bug ticket to fix both. In the end the fix is going to be the same one, or almost the same one.

**Comment by DavidJGapCR** [comment]
Looks good now.

**Review by DavidJGapCR — APPROVED** [review]

**Pull request merged**